gc/src/gc.rs

```diff
@@ -112,8 +112,7 @@ impl<T: Trace + ?Sized> GcBox<T> {
     pub unsafe fn root_inner(&self) {
         // abort if the count overflows to prevent `mem::forget` loops that could otherwise lead to
         // erroneous drops
-        self.header.roots.set(self.header.roots.get()
-                              .checked_add(1).unwrap_or_else(|| ::std::intrinsics::abort()));
+        self.header.roots.set(self.header.roots.get().checked_add(1).unwrap());
```

Comment:
Without explicitly looking into this, `abort` probably generates less code and is more easily optimizable than `unwrap`. It's also the same approach that `std` takes with `{Arc, Rc}`.